src/bootstrap/dylib_util.rs

```diff
@@ -26,3 +26,23 @@ pub fn dylib_path() -> Vec<PathBuf> {
     };
     env::split_paths(&var).collect()
 }
+
+lazy_static::lazy_static! {
+    static ref LLD_VERSION_OUT: String =
+        output(Command::new("lld").arg("-flavor").arg("ld").arg("--version"));
+    static ref LLD_VERSION: i32 = if let (Some(b), Some(e)) =
+        (LLD_VERSION_OUT.find(char::is_numeric), LLD_VERSION_OUT.find('.'))
+    {
+        LLD_VERSION_OUT.as_str()[b..e].parse::<i32>().ok().unwrap_or(14)
+    } else {
+        14
+    };
+}
+
+pub fn lld_flag_no_threads(is_windows: bool) -> String {
```

Comment:
I think a single lazy `static` keeping the `&'static str` result of this function should be enough.
It can replace both `LLD_VERSION_OUT`, `LLD_VERSION` and `lld_flag_no_threads`.
`cfg!(windows)` is a constant.